File: haskell/problem1.hs
-- Problem1.hs
module Problem1 (verificarEdad) where

verificarEdad :: Int -> String
verificarEdad edad
    | edad < 0        = "Error: edad no valida"
    | edad > 120      = "Error: edad no valida"
    | edad >= 18      = "Mayor de edad"
    | otherwise       = "Menor de edad"

-- funcion principal de prueba
main :: IO ()
main = do
    print (verificarEdad 20)    -- "Mayor de edad"
    print (verificarEdad 17)    -- "Menor de edad"
    print (verificarEdad (-5))  -- "Error: edad no valida"
    print (verificarEdad 150)   -- "Error: edad no valida"

File: python/problem1.py
# problem1.py
def verificar_edad(edad: int) -> str:
    """
    Verifica si la edad ingresada corresponde a un mayor o menor de edad.
    """
    if edad < 0 or edad > 120:
        return "Error: edad no valida"
    return "Mayor de edad" if edad >= 18 else "Menor de edad"

if __name__ == "__main__":
    try:
        edad = int(input("Ingresa tu edad: "))
        print(verificar_edad(edad))
    except ValueError:
        print("Error: debes ingresar un nmero entero.")

